### ISSUE TITLE
Python 3.12 incompatibility: "ModuleNotFoundError: No module named 'imp'" while building subunit dependency using Conan

### ISSUE THREAD

**Issue by vsbogd** [trueagi-io/hyperon-experimental#584]
`hyperonc` module is affected by https://bugs.launchpad.net/subunit/+bug/2052948

When trying to build it in environment with Python 3.12 the following error happens:
```
...
libtool: install: chmod 644 ~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378//lib/libcppunit_subunit.a
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'imp'
libtool: install: ranlib ~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378//lib/libcppunit_subunit.a
make[1]: *** [Makefile:957: install-pkgpythonPYTHON] Error 1
make[1]: *** Waiting for unfinished jobs....
libtool: warning: remember to run 'libtool --finish //lib'
make[1]: Leaving directory '~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/src'
make: *** [Makefile:1447: install-am] Error 2
ERROR: subunit/1.4.0: Error in package() method, line 147
 autotools.install()
 ConanException: Error 2 while executing make install 'INSTALLARCHLIB=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/lib' 'INSTALLSITEARCH=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/archlib' 'INSTALLVENDORARCH=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/archlib' 'INSTALLSITEBIN=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/bin' 'INSTALLSITESCRIPT=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/bin' 'INSTALLSITEMAN1DIR=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/share/man/man1' 'INSTALLSITEMAN3DIR=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/share/man/man3' DESTDIR=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378 -j16
```

**Comment by gl-yziquel** [comment]
This issue does not seem completely solved:

https://github.com/trueagi-io/hyperon-experimental/issues/703

The fix was, with pyenv, to ensure that conan-io was running in a 3.11 and not 3.12 environment.

This occured on git HEAD.

I'd suggest notifying the reader of the README.md file that 3.12 should be avoided.

**Comment by vsbogd** [comment]
The issue is a [subunit bug](https://bugs.launchpad.net/subunit/+bug/2052948) which seems to be fixed in the latest 1.4.3 version. Unfortunately it is not uploaded to Conan Center.
